ballista/core/src/replicator/mod.rs

```diff
@@ -0,0 +1,302 @@
+use std::sync::Arc;
+
+use bytes::Bytes;
+
+use datafusion::arrow::ipc::writer::{IpcWriteOptions, StreamWriter};
+use datafusion::arrow::ipc::CompressionType;
+use datafusion::arrow::record_batch::RecordBatch;
+use object_store::{path::Path, ObjectStore};
+use tokio::io::AsyncWriteExt;
+use tokio::{fs::File, sync::mpsc};
+use tracing::{info, warn};
+
+use crate::error::BallistaError;
+use tokio_util::compat::{Compat, TokioAsyncReadCompatExt};
+
+use self::async_reader::AsyncFileReader;
+pub mod async_reader;
+
+pub enum Command {
+    Replicate { job_id: String, path: String },
+}
+
+pub async fn start_replication(
+    base_path: String,
+    object_store: Arc<dyn ObjectStore>,
+    mut receiver: mpsc::Receiver<Command>,
+) -> Result<(), BallistaError> {
+    while let Some(Command::Replicate { job_id, path }) = receiver.recv().await {
```

Comment:
Can we add a few promehteus metrics here:

1. Number of replication events processed
2. Failed replications by reason (failed to parse path, failed to read file, failed to write)

Reply:
Good idea, done!